### PR TITLE
waitFor method support

### PR DESCRIPTION
**Pull request by polevaultweb**
Resolves https://github.com/WhatDaFox/Codeception-Mailtrap/issues/26

### CODE REVIEW

**Comment by polevaultweb** [comment]
@foxted I might need to change this, as its tightly coupled to phpunit 5.x

**Comment by foxted** [comment]
Yes, ideally, I'd like to keep the current dependencies. But I really like this feature idea @polevaultweb, thanks for the initiative :)

**Comment by polevaultweb** [comment]
@foxted I obviously didn't have enough coffee. Fixed up!

File: src/MailtrapWait.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	/**
+	 * Calls the function provided with the driver as an argument until the return value is not falsey.
+	 *
+	 * @param callable $func_or_ec
+	 * @param string $message
+	 *
+	 * @throws \Exception
+	 * @return mixed The return value of $func_or_ec
+	 */
+	public function until($func_or_ec, $message = '')
```
Comment (foxted) [inline src/MailtrapWait.php]: @polevaultweb I think we could make this a little clearer when it comes to naming variables. Wanna give it a shot?

File: src/MailtrapWait.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	/**
+	 * @var int
+	 */
+	protected $timeout;
+	/**
+	 * @var int
+	 */
+	protected $interval;
+
+	public function __construct(Mailtrap $mailtrap, $timeout_in_second = null, $interval_in_millisecond = null)
```
Comment (foxted) [inline src/MailtrapWait.php]: Small detail, but it's missing a docblock :)

**Review by foxted — CHANGES_REQUESTED** [review]

**Comment by polevaultweb** [comment]
@foxted how's this?

**Review by foxted — APPROVED** [review]

**Pull request merged**